### ISSUE TITLE
how to default to rendering only the end of a list of comments?

### ISSUE THREAD

**Issue by propertunist** [hypeJunction/hypeLists#23]
the default view for comments in elgg is to have the newest at the top and oldest at the bottom. this means that we are reading the thread in a way that is backwards to the usual way of reading text.
there is an option in elgg_list_entities called ''reverse_order_by' that solves that problem, but when i use it with hypeLists i get inconsistent results across my servers and also unexpected placing of the 'load more' buttons. i don't know why i am seeing inconsistent results across my servers, but what i would like to have is:

* newest comments at the bottom
* the visible list of comments being limited in size
* the 'load more' button being at the top of the list of comments, so that older comments can be loaded as desired.

so far i have tried adding the following parameters to elgg_list_entities, but i haven't found a combination that works:

* data-reversed
* data-paginationPosition
* position
* reverse_order_by

i have seen that in some cases i get a 'load more' button appear at both the top and the bottom of the list, but only when the page was opened via a river link for a mid level comment in the thread. this does not always work though as sometimes i click such a link and i just see the standard set of comments that i would see anyway when viewing the entity through a standard URL for it's page.

do you have any tips on how to achieve the desired outcome here?
thanks

**Comment by propertunist** [comment]
on the issue of clicking on a comment in the river - i find that when i put no special options for comments in the call to elgg_list_entities, i am taken to the middle of the list of comments (if the list is long) at an apprarently random point, which makes no reference to the position of the comment i clicked on in the river. this means i need to click 'back' up the list of comments using the ajax loader, to reach the comment that i want to read. 
it seems to me that maybe the comment system changed in some way in between hypeList being designed and now? is that right.. i'm not sure, but either way it is not working here as it needs to for ease of use.

**Comment by propertunist** [comment]
if i inspect the list container div in the browser, i see that no matter what i do, these attributes i am setting in the call to elgg_list_entities never make it into the container and thus never have any effect:
* data-offset
* data-reversed

**Comment by propertunist** [comment]
so i have got the reversed value to function correctly by just passing 'reversed' to elgg_list_entities rather than 'data-reversed' - but it was not the solution to the particular issue here anyway.

**Comment by hypeJunction** [comment]
Did you check hypeInteractions?

**Comment by propertunist** [comment]
no, not yet - i did look at it but not test it. i've already over-ridden many views and upgraded galliComments so i thought the simplest solution would be to just configure hypeLists how i need it to be. are you saying to look at the way hypeInteractions has used HypeLists? or to just use hypeInteractions?

**Comment by hypeJunction** [comment]
Either.  But you can compare how plugin settings affect list settings in hypeinteractions

**Comment by propertunist** [comment]
ok, so from looking here: https://github.com/hypeJunction/hypeInteractions/blob/8bb4bfbc8df0d8d2b29d78225222220a4f386893/views/default/framework/interactions/comments.php
it appears that i would just need to use the offset and maybe offset_key. does that sound correct?

**Comment by hypeJunction** [comment]
Sounds right.

**Comment by propertunist** [comment]
ok thanks, i'll take a look

**Comment by propertunist** [comment]
i have now coded the comments lists on my site to use an offset which is the total count of the comments, minus the limit for the list - which should render the final few comments and then allow the user to navigate back up to the older comments via ajax loads. unfortunately, although the correct comments are initially rendered, when i click the button to load the earlier comments - no comments are being loaded. you can see this here: https://www.ureka.org/thoughts/view/60766/internet

**Comment by hypeJunction** [comment]
What are the options for ege* in your list?

**Comment by propertunist** [comment]
i'm not sure what ege* is?

**Comment by hypeJunction** [comment]
elgg_get_entities_*

**Comment by propertunist** [comment]
ah.. 
```
 $content .= elgg_list_entities(array(
	'type' => 'object',
	'subtype' => 'comment',
	'container_guid' => $vars['entity']->guid,
	'reverse_order_by' => true,
	'full_view' => true,
	'limit' => $limit,
	'preload_owners' => true,
	'distinct' => false,
	'url_fragment' => $attr['id'],
  'auto_refresh' => 20,
	'pagination' => 'true',
	'pagination_type' => 'infinite',
	'reversed' => true,
	'list_id' => 'comments_list',
	'show_count' => false,
	'offset' => $count - $limit,
	'offset_key' => 'offset',
	'paginationPosition' => 'before',
    ));
```
i have just tested with a base_url, which made no difference and also switched to default pagination type. in that mode the correct number of page links were rendered, but when i click on one prior to the last one, the result is that no comments are shown - so the page is empty.

**Comment by hypeJunction** [comment]
What's the URL on the Load previous button?
Try hanging offset to 
`'offset' => get_input('offset', $count - $limit)`

**Comment by propertunist** [comment]
on my local installation, the url on the load previous button is:
```
http://mysite.org/videos/watch/6611?limit=11&offset=15
```
that's with a data count of 40, an offset of 29 and a limit of 11

**Comment by hypeJunction** [comment]
Try my above suggestion. Your offset is always set to the last page.

**Comment by propertunist** [comment]
i just tested with the amended offset value you posted and that has mostly fixed the issue! thanks! there are still some other, related minor issues though. e.g. on some pages the offset is not being calculated correctly and i end up with only 2 comments initially visible. also, sometimes the list of pages for normal pagination is rendered and not replaced by the 'load more' button (this is intermittent and probably not related to this current github issue though).

**Comment by hypeJunction** [comment]
You should probably add some logic before setting an offset. Check if $count > $limit, otherwise you end up with a negative offset.

**Comment by propertunist** [comment]
sure ok, good point - that's now done, though it didn't resolve the issue of the offset being miscalculated in my examples here as the count of the list of comments is higher than the limit value.

**Comment by hypeJunction** [comment]
Try removing 'distinct'

**Comment by propertunist** [comment]
ok, i did that but it made no difference. the offset that is being passed in as an input is incorrect, however, the one that is calculated by $count - $limit is correct. i looked through the PHP for where the offset is coming from but it's not in the core elgg code or my code, so i think it is coming from hypelist somewhere, is that right?

**Comment by hypeJunction** [comment]
What happens with default limit of 10? I can't remember any more, but I would assume it's somewhere in the javascript. I am quite sure it was working as it should in hypeInteractions. I can't be bothered to dig through hypeLists code at the moment.

**Comment by propertunist** [comment]
limit of 10 makes no difference. for the moment i am content that the comments list is usable as it is.. i'll come back to it soon and look more closely at the JS.
